src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -184,19 +185,47 @@ object HttpGoogleServicesDAO extends GoogleServicesDAO with FireCloudRequestBuil
       .build()
   }
 
-  def getRawlsServiceAccountAccessToken = {
-    val googleCredential = new GoogleCredential.Builder()
+  def getRawlsServiceAccountCredential: Credential = {
+    new GoogleCredential.Builder()
       .setTransport(httpTransport)
       .setJsonFactory(jsonFactory)
       .setServiceAccountId(rawlsPemFileClientId)
       .setServiceAccountScopes(storageReadOnly)
       .setServiceAccountPrivateKeyFromPemFile(new java.io.File(rawlsPemFile))
       .build()
+  }
+
+  def getRawlsServiceAccountAccessToken = {
+    val googleCredential = getRawlsServiceAccountCredential
 
     googleCredential.refreshToken()
     googleCredential.getAccessToken
   }
 
+  override def listObjectsAsRawlsSA(bucketName: String, prefix: String): List[String] = {
+    val storage = new Storage.Builder(httpTransport, jsonFactory, getRawlsServiceAccountCredential).setApplicationName(appName).build()
+    val listRequest = storage.objects().list(bucketName).setPrefix(prefix)
+    Try(executeGoogleRequest[Objects](listRequest)) match {
+      case Failure(ex) =>
+        // TODO: do anything else here?
```

Comment:
well, is there?

Reply:
also you're using a `Try` and then discarding it in favour of throwing an exception -- might as well return the `Try` from the function

Reply:
I don't want to return the `Try` because nothing else in this class does - they all throw exceptions. But I've made the exception case do something now.